lambdas/call-dcmaw-async-cri/src/test/java/uk/gov/di/ipv/core/calldcmawasynccri/CallDcmawAsyncCriHandlerTest.java

```diff
@@ -0,0 +1,167 @@
+import uk.gov.di.ipv.core.library.domain.ErrorResponse;
+import uk.gov.di.ipv.core.library.domain.ProcessRequest;
+import uk.gov.di.ipv.core.library.exceptions.SqsException;
+import uk.gov.di.ipv.core.library.persistence.item.ClientOAuthSessionItem;
+import uk.gov.di.ipv.core.library.persistence.item.IpvSessionItem;
+import uk.gov.di.ipv.core.library.service.ClientOAuthSessionDetailsService;
+import uk.gov.di.ipv.core.library.service.ConfigService;
+import uk.gov.di.ipv.core.library.service.IpvSessionService;
+import uk.gov.di.ipv.core.library.verifiablecredential.domain.VerifiableCredentialResponse;
+import uk.gov.di.ipv.core.library.verifiablecredential.domain.VerifiableCredentialStatus;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.di.ipv.core.library.domain.CriConstants.DCMAW_ASYNC_CRI;
+
+@ExtendWith(MockitoExtension.class)
+class CallDcmawAsyncCriHandlerTest {
+    public static final String TEST_USER_ID = "a-user-id";
+    public static final String TEST_OAUTH_STATE = "some_dummy_state_value";
+    public static final ClientOAuthSessionItem clientOAuthSessionItem =
+            ClientOAuthSessionItem.builder()
+                    .userId(TEST_USER_ID)
+                    .clientOAuthSessionId(TEST_OAUTH_STATE)
+                    .govukSigninJourneyId("a-govuk-journey-id")
+                    .build();
+    private static final ProcessRequest input =
+            ProcessRequest.processRequestBuilder()
+                    .ipvSessionId("a-session-id")
+                    .ipAddress("an-ip-address")
+                    .clientOAuthSessionId("an-oauth-session-id")
+                    .journey("a-journey")
+                    .lambdaInput(Map.of("journeyType", "ipv"))
+                    .build();
+    public static final String JOURNEY_ENHANCED_VERIFICATION = "/journey/enhanced-verification";
+    @Mock private Context mockContext;
+    @Mock private ConfigService mockConfigService;
+    @Mock private IpvSessionService mockIpvSessionService;
+    @Mock private ClientOAuthSessionDetailsService mockClientOAuthSessionDetailsService;
+    @Mock private DcmawAsyncCriService mockDcmawAsyncCriService;
+    @Mock private CriStoringService mockCriStoringService;
+    @Mock private IpvSessionItem mockIpvSessionItem;
+    @Mock private VerifiableCredentialResponse mockVerifiableCredentialResponse;
+    @InjectMocks private CallDcmawAsyncCriHandler callDcmawAsyncCriHandler;
+
+    @BeforeEach
+    public void setUp() {
+        mockIpvSessionItem.setIpvSessionId("a-session-id");
+    }
+
+    @Test
+    void handleRequestShouldCallDcmawCriAndReturnJourneyNextIfNoBreachingCiReceived()
```

Comment:
I don't think we test for breaching CIs as we only ever expect a pending VC back?